homeassistant/components/energy/data.py

```diff
@@ -79,7 +79,16 @@ class SolarSourceType(TypedDict):
     config_entry_solar_forecast: list[str] | None
 
 
-SourceType = Union[GridSourceType, SolarSourceType]
+class BatterySourceType(TypedDict):
+    """Dictionary holding the source of battery storage."""
+
+    type: Literal["battery"]
+
+    stat_energy_from: str
+    stat_energy_to: str
```

Comment:
It seems like batteries can both go to the home and to the grid.

Reply:
Yes, but that will be represented by the same entity. It left the battery, where it then went doesn't matter for the battery.